qt/widgets/instrumentview/src/InstrumentWidgetMaskTab.cpp

```diff
@@ -487,10 +510,131 @@ void InstrumentWidgetMaskTab::setActivity() {
     m_instrWidget->getSurface()->setInteractionMode(
         ProjectionSurface::DrawFreeMode);
     m_activeTool->setText("Tool: Free draw. " + whatIsBeingSelected);
+  } else if (m_pixel->isChecked()) {
+    m_activity = Pixel;
+    m_instrWidget->getSurface()->setInteractionMode(
+        ProjectionSurface::PickSingleMode);
+    m_activeTool->setText("Tool: Pixel mask. " + whatIsBeingSelected);
+  } else if (m_tube->isChecked()) {
+    m_activity = Tube;
+    m_instrWidget->getSurface()->setInteractionMode(
+        ProjectionSurface::PickTubeMode);
+    m_activeTool->setText("Tool: Tube/bank mask. " + whatIsBeingSelected);
   }
   m_instrWidget->updateInfoText();
 }
 
+/**
+ * Slot responding on the pick of a single detector pixel for masking.
+ * Can be used either to mask the picked pixel or its parent.
+ */
+void InstrumentWidgetMaskTab::singlePixelPicked(size_t pickID) {
+  auto &actor = m_instrWidget->getInstrumentActor();
+  const auto &componentInfo = actor.componentInfo();
+  if (!componentInfo.isDetector(pickID)) {
+    return;
+  }
+
+  QApplication::setOverrideCursor(QCursor(Qt::WaitCursor));
+  std::vector<size_t> detectorId{pickID};
+  m_instrWidget->updateInstrumentView(); // to refresh the pick image
+  auto wsMask = actor.getMaskWorkspace();
+
+  // two cases : either we are on roi mode, or on masking mode
+  if (m_masking_on->isChecked()) {
+    // then either we are on single pixel picking mode, on on tube picking mode
+    if (m_pixel->isChecked()) {
+      actor.addMaskBinsData(detectorId);
+
+      Mantid::detid_t detId = actor.getDetID(pickID);
+
+      // try to mask the detector, ignore any failure
+      try {
+        wsMask->setMasked(detId);
+      } catch (...) {
```

Comment:
I think instead of catching all, it would be better to catch `std::runtime_error` and log the error message in debug channel. There are few other instances of this.